app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedPresenter.java

```diff
@@ -40,45 +45,75 @@
 
     @Inject
     public NewsFeedPresenter() {
+        disposables = new CompositeDisposable();
     }
 
     @Override
-    public void start() {
+    public void subscribe() {
         Log.d(TAG, "start: " + mainApplication);
         Category.resolveCategory(mainApplication, Arrays.asList(Category.values()));
         initStartValues();
         updateActualNews();
     }
 
+    @Override
+    public void unsubscribe() {
+        disposables.clear();
+    }
+
     @Override
     public void updateActualNews() {
-        view.showProgressBar();
-        updateNewsStorage();
+        NewsFeedParams params = new NewsFeedParams(new Filter(category, showOnlyFavorite));
+        Disposable subscription = updateStorageUseCase.run(params)
+                .flatMap(updatedNews -> getFilteredNewsUseCase.run(params))
+                .doOnSubscribe(disposable -> view.showProgressBar())
+                .onErrorResumeNext(throwable -> getFilteredNewsUseCase.run(params))
+                .subscribeOn(Schedulers.computation())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        resultNews -> { /* Emit */
+                            setCashedNews(resultNews);
+                            updateNewsListUI(resultNews);
+                            view.hideProgressBar();
+                            view.hideRefreshLayout();
+                        },
+                        throwable -> { /* Error */
+                            Log.e(TAG, "updateActualNews failure: " + throwable);
```

Comment:
Don't you want to call `view.hideProgressBar();` and `view.hideRefreshLayout();` on error? It might happen in case both network and database calls will fail.